src/rules/LinearAlgebra/blas.jl

```diff
@@ -0,0 +1,115 @@
+module BLASRules
+
+using ..Enzyme
+using LinearAlgebra.BLAS
+
+const ConstOrDuplicated{T} = Union{Const{T},Duplicated{T}}
+
+_safe_similar(x::AbstractArray, n::Integer) = similar(x, n)
```

Comment:
These utilities should be reusable for and greatly simplify the rules for all other Level 1 BLAS functions.